File: tesonet_party/app/src/main/java/lt/liutkevicius/tesonetandroidparty/ui/base/BasePresenter.java
package lt.liutkevicius.tesonetandroidparty.ui.base;

import io.reactivex.disposables.CompositeDisposable;

public class BasePresenter<T> implements Presenter<T> {

    protected final CompositeDisposable subscriptions = new CompositeDisposable();
    private T view;

    public boolean hasView() {
        return view != null;
    }

    public T getView() {
        return view;
    }

    @Override
    public void setView(T view) {
        this.view = view;
        if (view == null) {
            subscriptions.clear();
        }
    }
}

File: app/src/main/java/levkovskiy/dev/tesonettest/mvp/presenter/BasePresenter.kt
package levkovskiy.dev.tesonettest.mvp.presenter

import io.reactivex.disposables.CompositeDisposable
import io.reactivex.disposables.Disposable

open class BasePresenter {

  private val disposables = CompositeDisposable()

  protected fun unsubscribeOnDestroy(disposable: Disposable) {
    disposables.add(disposable)
  }


}